src/settings.py

```diff
@@ -130,6 +130,7 @@ class GlobalSettings:
     grammar_search_heuristic_based_max_demos = 5
     grammar_search_lookahead_based_temperature = 10.
     grammar_search_task_planning_timeout = 1.0
+    grammar_search_hill_climbing_depth = 0
```

Comment:
is there a reason we are keeping this at 0? wouldn't it be strictly better to have it higher since otherwise we'd just terminate and be sad? (unless having it higher is causing us to add _more_ predicates once we already have the ideal set......)

Reply:
setting it to 1 slows everything down very significantly